src/components/Tab.vue

```diff
@@ -1,6 +1,19 @@
 <template>
-    <div class="tab">
-        <div class="drag"></div>
+    <div class="tab" @click="selected" last-coords-x="0" last-coords-y="0">
```

Comment:
both tabs (visualizer and sequence) have the default "last undocked position" set to 0,0.  This means if you open up a fresh scope and immediately use the tab buttons to undock both tabs, they cover each other up in the upper left, which could be confusing.  It would probably be better to set one of them to default lower down.

Reply:
OK, if @kaldis-berzins or colleagues don't have a chance to add a commit to this PR handling this point, I will take a crack at it when next I have the opportunity.

Reply:
Another corner case behaviour related to this.  If you open a fresh scope and then move both tabs from docked-right to docked-left, they also keep their "last docked" variable set to upper left (0,0) and are still registered as docked.  So then when you click the "dock/undock" button they don't return to the right side (as they would do if you moved both tabs to the middle).  They just go "up" on top of each other.  So given this, I actually think the best solution is for the `last-coords-x/y` variables to be defaulting to something in the middle of the screen, not top and bottom upper left as my original comment may have suggested.

Reply:
Maybe sort of one tab width to the right of the left edge? With Sequence at the top edge and Visualizer near the bottom?

Reply:
Here's another oddity that doesn't feel right: If you load the default visualizer per the code of this PR ("Random visualized by ModFill"), and then click the minimize button on the Visualizer tab, and then immediately click the "unminimize" button on Visualizer, it does not go back to the same size that it was -- after this operation Sequence is taller and Visualizer is shorter than at the beginning. Is that sufficiently poor behavior that we would like to address it at the level of this PR?

Reply:
> Maybe sort of one tab width to the right of the left edge? With Sequence at the top edge and Visualizer near the bottom?

Maybe slightly more than one tab width, and slightly down from the top edge.  This failure to align/tile would make the visual impression that we are "floating" more immediate.  Also, we could have the second one float similarly, but slight further down and right, so they are overlapping a little.  That creates a strong visual impression of floating boxes.

Reply:
Another observation: You cannot simultaneously minimize both Sequence and Visualizer. when docked. When one is minimized, the other is maximized to take the whole dock area. While ultimately that also does not feel "right" to me and I think we should revisit that behavior at some point, at the level of this PR I think it is the intended design choice and we should not seek to do anything about it in this current review. If anyone disagrees, please comment.

Reply:
The problem with my suggestion being, of course, that the more we move them off the top left, the more it becomes a potential problem on a small screen.

Reply:
> The problem with my suggestion being, of course, that the more we move them off the top left, the more it becomes a potential problem on a small screen.

How about Sequence a half tab width down and to the right from the top left, and Visualizer a full tab width down and to the left?

Reply:
> Here's another oddity that doesn't feel right: If you load the default visualizer per the code of this PR ("Random visualized by ModFill"), and then click the minimize button on the Visualizer tab, and then immediately click the "unminimize" button on Visualizer, it does not go back to the same size that it was -- after this operation Sequence is taller and Visualizer is shorter than at the beginning. Is that sufficiently poor behavior that we would like to address it at the level of this PR?

I'd leave this in the category of "not important enough for slowing the PR chain".  Unless it's super easy/obvious to fix (which may be the case), in which case the slowing is negligible.  If we don't fix it, we should log an issue, but I don't think it is likely to cause much consternation compared to many other outstanding issues.

Reply:
> > The problem with my suggestion being, of course, that the more we move them off the top left, the more it becomes a potential problem on a small screen.
> 
> How about Sequence a half tab width down and to the right from the top left, and Visualizer a full tab width down and to the right?

Here's a screenshot of this proposal (for "undocking" of both tabs immediately upon startup).
![Screenshot_2024-06-28_17-28-00](https://github.com/numberscope/frontscope/assets/3825429/80368293-6ff2-4663-8760-e0b4eebd1e35)

Reply:
> > The problem with my suggestion being, of course, that the more we move them off the top left, the more it becomes a potential problem on a small screen.
> 
> How about Sequence a half tab width down and to the right from the top left, and Visualizer a full tab width down and to the right?

If you're trying it out, I trust your instinct.  That sounds reasonable.

Reply:
> Another observation: You cannot simultaneously minimize both Sequence and Visualizer. when docked. When one is minimized, the other is maximized to take the whole dock area. While ultimately that also does not feel "right" to me and I think we should revisit that behavior at some point, at the level of this PR I think it is the intended design choice and we should not seek to do anything about it in this current review. If anyone disagrees, please comment.

I agree.

Reply:
> > > The problem with my suggestion being, of course, that the more we move them off the top left, the more it becomes a potential problem on a small screen.
> > 
> > 
> > How about Sequence a half tab width down and to the right from the top left, and Visualizer a full tab width down and to the right?
> 
> Here's a screenshot of this proposal (for "undocking" of both tabs immediately upon startup). !

Oh that looks nice.  Just to clarify, I think you mean it still starts  up docked, but if you immediately undock, you get this.

Reply:
I assume the first one you undock is at back in the overlap?

Reply:
Correct on both counts. I basically know how to do this now, but I still don't know how to make it that the coordinates are tied with the tab width so that they are intentionally 1/2 and one full tab width, respectively, as opposed to by coincidence being those amounts. So I need to figure out how to get the values of constants into the CSS _and_ into the component templates.

However, in trying this out I discovered another disturbing behavior: if you doc a tab to the left, and then undock it and immediately re-dock it, it docks to the right. That does not seem right. As with the others, is it sufficiently "un-right" that we should address it in this PR? It's the most "bug-like" of these docking behaviors that I have noticed so far.

Reply:
About left/right docking.  I noticed that behaviour but I assumed it was intentional.  I suppose however, it would be better to dock to wherever you were docked last.  Pondering it a bit more, I think it would be good to fix this.  It's a bit finnicky, though... You can actually dock one on the left and one on the right, so it's a per-tab variable that would be needed.

Reply:
That's not a problem, the last undocked coordinates are already per tab so no reason why the last dock position can't be.

Reply:
OK, I have pushed a commit that implements those initial positions and also returns tabs to the last place they were docked when you hit the dock button. Please try it out and if it all looks good, please merge this PR! (I am happy with it at this point.)